python/nav/web/status2/views.py

```diff
@@ -47,6 +47,13 @@ def get_status_preferences(self):
         )
         if preferences:
             try:
+                if preferences.startswith("b'") and preferences.endswith("'"):
+                    # Status filter was incorrectly saved as a string representation of
+                    # bytes without being decoded ("b'1234", instead of "1234")
+                    preferences = preferences[2:-1]
+                    self.request.account.preferences[
+                        self.request.account.PREFERENCE_KEY_STATUS
+                    ] = preferences
```

Comment:
I'm not sure what these lines accomplish. I t looks like you're maybe trying to update the database record, but nothing is saved, so the update won't be persisted.

Not that it really matters, because the PR as a whole will ensure the value is serialized correctly next time the preferences are changed and saved.

You can either
1. Save the preferences at this point, or
2. just remove these lines, as it doesn't matter all that much...